Allow the observer to be deployed independently from Osiris

CermakM opened the pull request:
Observer can now pass kubernetes event to Osiris directly

- get rid of Osiris dependency

Commit:
Observer can now pass kubernetes event to Osiris directly

- get rid of Osiris dependency

Signed-off-by: Marek Cermak <macermak@redhat.com>

modified:   app.py
modified:   Pipfile
modified:   Pipfile.lock
modified:   requirements.txt

Commit:
Modify app.py header

Signed-off-by: Marek Cermak <macermak@redhat.com>

modified:   app.py

CermakM commented:
Addresses #3

ghost commented:
Build failed.

- [thoth-coala ](https://zuul.thoth-station.ninja/logs/7/7/9b77e4cc82714a7905ca913caf9b422dfa9ca0a5/check/thoth-coala/053343e/) : RETRY_LIMIT in 1m 46s

goern commented:
recheck

ghost commented:
Build failed.

- [thoth-coala ](https://zuul.thoth-station.ninja/logs/7/7/9b77e4cc82714a7905ca913caf9b422dfa9ca0a5/check/thoth-coala/80e89f2/) : FAILURE in 57s

Commit:
Use Thoth common OpenShift and loggin API

- replace kubernetes client with OpenShift client
- use ResourceInstance instead of kuberenetes V1Event (NOTE: This causes
temporarily incompatibility with Osiris API)
- do not require additional event filters (only get build resources)

Signed-off-by: Marek Cermak <macermak@redhat.com>

modified:   Pipfile
modified:   Pipfile.lock
modified:   app.py
modified:   openshift/deployment-template.yaml
modified:   openshift/template.yaml

ghost commented:
Build failed.

- [thoth-coala ](https://zuul.thoth-station.ninja/logs/7/7/be40a8cb22dddc107ddc6499811a4458018633c5/check/thoth-coala/9f51b14/) : FAILURE in 57s

goern commented on app.py:
```diff
@@ -43,78 +41,36 @@
-_SERVICE_CERT_FILENAME = "/var/run/secrets/kubernetes.io/serviceaccount/ca.crt"
-
-if Path(_SERVICE_TOKEN_FILENAME).exists():  # in-cluster configuration
+    _NAMESPACE = os.getenv("MIDDLETIER_NAMESPACE", None)
```
is this indentation level correct?

CermakM replied:
Think it was, but it is no longer applicable anyway, in the next commit I am gonna get rid of all that.

Commit:
Use Thoth common library wherever possible

- replace logging with thoth.common logging
- replace kubernetes client with Thoth OpenShift API

- also put logs to the dedicated Osiris endpoint

Signed-off-by: Marek Cermak <macermak@redhat.com>

modified:   app.py

ghost commented:
Build failed.

- [thoth-coala ](https://zuul.thoth-station.ninja/logs/7/7/e27b0c0b3151872d0d6265ec39083e6e91947d3b/check/thoth-coala/10557dd/) : FAILURE in 37s

Commit:
Fix issues reported by coala

Signed-off-by: Marek Cermak <macermak@redhat.com>

modified:   app.py
modified:   openshift/template.yaml

ghost commented:
Build failed.

- [thoth-coala ](https://zuul.thoth-station.ninja/logs/7/7/9b9d2516d6a6b25e403b2c94cc2d1af05806f822/check/thoth-coala/98108fe/) : FAILURE in 50s

ghost commented:
Build failed.

- [thoth-coala ](https://zuul.thoth-station.ninja/logs/7/7/02f18e5823f0612c0046a9c52f7f9a1c02298f17/check/thoth-coala/e738451/) : FAILURE in 1m 04s
